### ISSUE TITLE
How do I use Actions.reset() method?

### ISSUE THREAD

**Issue by xkawi** [aksonov/react-native-router-flux#23]
Hi,

Thanks for the great work.

My intention is that after a few push navigations that result to stack like this: `[page1, page2, page3]`, and upon clicking a button on page3 I want to show page1 and reset the stack to: `[page1]`

I have tried these methods:
1. on page3 render method: `<Button onPress={()=>Actions.reset()}>OK</Button>`
2. on page3 render method: `<Button onPress={()=>Actions.reset('page1')}>OK</Button>`
3. on page1(initial page): Add `type='reset'` attribute for page1 Route defined in the main source code (index.ios.js)

However all methods result in this error:
![screen shot 2015-10-17 at 11 00 45 pm](https://cloud.githubusercontent.com/assets/7726233/10559408/f85fe098-7522-11e5-900c-d5a81a2d8ec4.png)

Here is the debug console from chrome:
![screen shot 2015-10-17 at 11 15 36 pm](https://cloud.githubusercontent.com/assets/7726233/10559529/1f007e0e-7525-11e5-96fb-c54717cb5eff.png)

It seems that `page.initial = undefined` hence unable to find the route, could this be the cause? if yes, how should I rectify it? if not, which part am I doing wrong?
let me know if you need more information.
thank you.

PS: I installed version 0.3.4 if that helps.

**Comment by aksonov** [comment]
Thanks for detailed info, seems to be bug, should be page.name instead of page.initial, could you check and if it works, commit PR?

**Comment by xkawi** [comment]
Just realised it was my mistake. I forget to remove `Actions.reset()` on `componentDidMount()` in page1 that I added for debugging purpose earlier. I re-installed the library and try using the reset method again just to verify it has no issue, and it works. So, there is no bug with the library as of now.

fyi, I observed 2 things:

1. `Actions.reset()` will results in the error mentioned earlier (`no route for data:{}`), but `Actions.reset('page1')` works as expected. Is this the intended behaviour? or `Actions.reset()` should works and automatically reset back to the initial page?

2. If I specify another page like `Actions.reset('page2')` it works too, although page2 is not the initial page that I defined. If 'page2' has a back button in the navigation bar or anything else, it will appear. Is this the expected behaviour? this might be related to the first observation though.

Out of curiosity, I tried your recommendation and change to `page.name`. I called the reset method this way `Actions.reset('page1')` and found that `page.name` always equals to the current route 'page3' in this case. This further verify that there is no bug in your code, unless my observation says otherwise.

hope this helps. closing this comment as it has been resolve.
feel free to open it when needed. thanks.
